scripts/zones/Aht_Urhgan_Whitegate/npcs/Rytaal.lua

```diff
@@ -1,93 +1,106 @@
+    {
+        [1] = {KI = tpz.keyItem.LEUJAOAM_ASSAULT_ORDERS, points = "LEUJAOAM_ASSAULT_POINT"},
+        [2] = {KI = tpz.keyItem.MAMOOL_JA_ASSAULT_ORDERS, points = "MAMOOL_ASSAULT_POINT"},
+        [3] = {KI = tpz.keyItem.LEBROS_ASSAULT_ORDERS, points = "LEBROS_ASSAULT_POINT"},
+        [4] = {KI = tpz.keyItem.PERIQIA_ASSAULT_ORDERS, points = "PERIQIA_ASSAULT_POINT"},
+        [5] = {KI = tpz.keyItem.ILRUSI_ASSAULT_ORDERS, points = "ILRUSI_ASSAULT_POINT"},
+        [6] = {KI = tpz.keyItem.NYZUL_ISLE_ASSAULT_ORDERS, points = "NYZUL_ISLE_ASSAULT_POINT"},
+    }
 
-function onTrade(player,npc,trade)
+function onTrade(player, npc, trade)
 end
 
-function onTrigger(player,npc)
-
-    local currentday = tonumber(os.date("%j"))
-    local lastIDtag = player:getCharVar("LAST_IMPERIAL_TAG")
+function onTrigger(player, npc)
+    local lastIDtag = player:getCharVar("lastTagTime")
     local tagCount = player:getCurrency("id_tags")
-    local diffday = currentday - lastIDtag
     local currentAssault = player:getCurrentAssault()
-    local haveimperialIDtag
+    local tagsAvail = 0
 
     if player:getCurrentMission(TOAU) == tpz.mission.id.toau.PRESIDENT_SALAHEEM and player:getCharVar("AhtUrganStatus") == 0 then
         player:startEvent(269,0,0,0,0,0,0,0,0,0)
     elseif player:getCurrentMission(TOAU) <= tpz.mission.id.toau.IMMORTAL_SENTRIES or player:getMainLvl() <= 49 then
         player:startEvent(270)
-    elseif currentAssault ~= 0 and tpz.besieged.hasAssaultOrders(player) == 0 then
+    elseif currentAssault ~= 0 and player:getCharVar("assaultEntered") ~= 0 then
         if player:getCharVar("AssaultComplete") == 1 then
             player:messageText(player,ID.text.RYTAAL_MISSION_COMPLETE)
             player:completeAssault(currentAssault)
+        elseif currentAssault == 51 then
+            player:messageText(player, ID.text.NYZUL_FAIL)
+            player:delAssault(currentAssault)
         else
-            player:messageText(player,ID.text.RYTAAL_MISSION_FAILED)
-            player:addAssault(0)
+            local assaultPoint = assaultOrders[player:getVar("assaultEntered")].points
+            player:addCurrency(assaultPoint, 100)
+            player:messageText(player, ID.text.ASSAULT_FAILED)
+            player:delAssault(currentAssault)
+        end
+        player:setCharVar("AssaultComplete", 0)
+        player:setCharVar("assaultEntered", 0)
+        player:setCharVar("Assault_Armband", 0)
+        for _, v in pairs(assaultOrders) do
+            if player:hasKeyItem(v.KI) then
+                player:delKeyItem(v.KI)
+            end
         end
-        player:setCharVar("AssaultComplete",0)
     elseif player:getCurrentMission(TOAU) > tpz.mission.id.toau.PRESIDENT_SALAHEEM or (player:getCurrentMission(TOAU) == tpz.mission.id.toau.PRESIDENT_SALAHEEM and player:getCharVar("AhtUrganStatus") >= 1) then
+        local lastTagTime = lastIDtag - 1009897200
+        local diffday =  math.floor((os.time() - lastIDtag)/86400)
```

Comment:
You could support the ROV KI here:
```
local mustWaitTime = 86400 -- 1 day in seconds
if player:hasKeyItem(tpz.ki.RHAPSODY_IN_AZURE) then mustWaitTime = 600 end -- set to 10 minutes
local diffday =  math.floor((os.time() - lastIDtag)/mustWaitTime)
```

Reply:
Yes there will be alot of changes to retail today configuration. I wrote this all for in era. I did add 1 settings so for for ENABLE_TOAU_ERA and set it for how many people can enter an assault. Era is 3 to 6. Retail is 1 to 6. But most of these changes will be done once I get it setup and working